core/src/main/java/uk/gov/gchq/koryphe/util/JavaUtils.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.koryphe.util;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * Copies of various methods/class from the Java 11 JDK.
+ */
+public final class JavaUtils {
+
+    private JavaUtils() {
+        // Empty
+    }
+
+    public static <T> T requireNonNullElse(final T input, final T other) {
+        if (null != input) {
+            return input;
+        } else {
+            return other;
+        }
+    }
+
+    public static class Set {
+
+        public static <T> HashSet<T> of(final T... item) {
+            return com.google.common.collect.Sets.newHashSet(item);
```

Comment:
The Javadoc for [`Set.of`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html#of(E...)) (Java 11 only) states:

>  Returns an **unmodifiable** set containing an arbitrary number of elements. 

Whereas the Javadoc for Guava [`com.google.common.collect.Sets.newHashSet`](url) states:

> Creates a **mutable** HashSet instance initially containing the given elements.

As such, I think wrapping the Guava method with `Collections.unmodifiableSet()` would be closer to how this works in Java 11. The same goes for `copyOf` below which should also return an unmodifiable set.